File: AlphaApi/Entities/Coffee.cs
namespace AlphaApi.Entities;

public class Coffee
{
    public int Id { get; set; }
    public required string Type { get; set; }
    public required string Shop { get; set; }
    public required string Location { get; set; }
    public int Intensity { get; set; }
    public int Rating { get; set; }
    public required string Temperature { get; set; }
    public string? Notes { get; set; }
    public int UserId { get; set; }
    public User User { get; set; } = null!;
}

File: alpha-api/src/schemas/coffee.py
from pydantic import BaseModel, ConfigDict, Field

from .users import UserOut


class CoffeeCreate(BaseModel):
    type: str
    shop: str
    location: str
    intensity: int = Field(ge=1, le=10)
    rating: int = Field(ge=1, le=5)
    temperature: str
    notes: str | None = None
    userId: int


class CoffeeUpdate(CoffeeCreate):
    pass


class CoffeeOut(BaseModel):
    model_config = ConfigDict(from_attributes=True)

    id: int
    type: str
    shop: str
    location: str
    intensity: int
    rating: int
    temperature: str
    notes: str | None = None
    userId: int = Field(validation_alias="user_id")


class CoffeeDetailOut(CoffeeOut):
    user: UserOut
